test/test_analyze.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright (c) 2025 National Land Survey of Finland (Maanmittauslaitos)
+#
+#  This file is part of geogen-algorithms.
+#
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+import pytest
+from geopandas import GeoDataFrame
+from shapely.geometry import Polygon
+
+from geogenalg.analyze import calculate_coverage
+
+
+def test_calculate_coverage():
+    calculate_coverage(
+        overlay_features=GeoDataFrame(
+            {
+                "id": [1],
+            },
+            geometry=[
+                Polygon([(0, 0), (4, 0), (4, 4), (0, 4), (0, 0)]),
+            ],
+            crs="EPSG:3067",
+        ),
+        base_features=GeoDataFrame(
+            {
+                "class": ["a"],
+                "id": [1],
+            },
+            geometry=[
+                Polygon([(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)]),
+            ],
+            crs="EPSG:3067",
+        ),
+    )
+
```

Comment:
Test is missing asserts, there should be at least two:
- assert that "coverage" is in resulting geodataframe columns
- assert that overlay_features["coverage"] has the correct value